### PR TITLE
Registration Form Fix

### PR DESCRIPTION
**Pull request by jmm8046**
Chaning registration resolver so that it can no longer assign to type RegstrationApiResolver

### CODE REVIEW

File: user-registration-app/src/app/models/registration.ts [hunk 1]
```diff
@@ -174,4 +174,38 @@ export class RegistrationApiResponse {
   isCurrentRegistration(): boolean {
     return this.hackathon.active;
   }
+  parseFromApiResponse(value: RegistrationApiResponse) : Registration{
```
Comment (sushshring) [inline user-registration-app/src/app/models/registration.ts]: This should be a static method under Registration, not RegistrationApiResponse

File: user-registration-app/src/app/services/resolvers/RegistrationResolver/registration.resolver.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class RegistrationResolver implements Resolve<Registration | Registration
             this.router.navigate([AppConstants.PIN_ENDPOINT]);
             return null;
           }
-          return registration;
+          return registration.parseFromApiResponse(registration);
```
Comment (sushshring) [inline user-registration-app/src/app/services/resolvers/RegistrationResolver/registration.resolver.ts]: return Registration.parseFromApiResponse(registration) after the method is made static

File: user-registration-app/src/app/models/registration.ts (hunk 1 above)
Comment (sushshring) [reply user-registration-app/src/app/models/registration.ts]: Also should be a public method.

File: user-registration-app/src/app/models/registration.ts [hunk 3]
```diff
@@ -174,4 +174,38 @@ export class RegistrationApiResponse {
...
+    registration.mlhdcp = !!value.mlh_dcp;
+    registration.ethnicity = value.race;
+    registration.codingExperience = value.coding_experience;
+    registration.referral = value.referral;
+    registration.projectDesc = value.project;
+    registration.expectations = value.expectations;
+    registration.veteran = value.veteran;
+    registration.submitted = !!value.submitted;
+    registration.hackathon = this.parseHackathon(registration);
+    registration.pin = (this.pinAi-value.pin).toString();
```
Comment (jmm8046) [inline user-registration-app/src/app/models/registration.ts]: @sushshring Once I move this up into Registration, this.pinAi doesn't exist anymore, I've been assuming that value.pin is the pin of the hackathon and not the pin of the user which is what registration.pin is meant to be. Is that right?

File: user-registration-app/src/app/models/registration.ts (hunk 1 above)
Comment (jmm8046) [reply user-registration-app/src/app/models/registration.ts]: done

File: user-registration-app/src/app/services/resolvers/RegistrationResolver/registration.resolver.ts (hunk 2 above)
Comment (jmm8046) [reply user-registration-app/src/app/services/resolvers/RegistrationResolver/registration.resolver.ts]: done

File: user-registration-app/src/app/models/registration.ts (hunk 1 above)
Comment (sushshring) [reply user-registration-app/src/app/models/registration.ts]: Still not static

File: user-registration-app/src/app/models/registration.ts [hunk 4]
```diff
@@ -56,10 +56,44 @@ export class Registration {
...
+    registration.mlhdcp = value.mlh_dcp;
+    registration.ethnicity = value.race;
+    registration.codingExperience = value.coding_experience;
+    registration.referral = value.referral;
+    registration.projectDesc = value.project;
+    registration.expectations = value.expectations;
+    registration.veteran = value.veteran;
+    registration.submitted = value.submitted;
+    registration.hackathon = this.parseHackathon(registration);
+    registration.pin = (value.pin).toString();
```
Comment (sushshring) [inline user-registration-app/src/app/models/registration.ts]: Nit: unnecessary parentheses

Comment (jmm8046) [reply user-registration-app/src/app/models/registration.ts]: done

**Pull request merged**